### PR TITLE
chore: bump various k8s.io components' version

### PR DESCRIPTION
**Pull request by nddq**
# Description

Dependabot is having issues with bumping multiple k8s.io components' version due to dependency issues. This PR consolidated all version bumps.

## Related Issue

If this pull request is related to any issue, please mention it here. Additionally, make sure that the issue is assigned to you before submitting this pull request.

## Checklist

- [ ] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [ ] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [ ] I have correctly attributed the author(s) of the code.
- [ ] I have tested the changes locally.
- [ ] I have followed the project's style guidelines.
- [ ] I have updated the documentation, if necessary.
- [ ] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed

Please add any relevant screenshots or GIFs to showcase the changes made.

## Additional Notes

Add any additional notes or context about the pull request here.

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+toolchain go1.22.2
 
 require (
 	github.com/go-chi/chi/v5 v5.0.11
 	github.com/google/uuid v1.6.0
 	github.com/prometheus/client_golang v1.19.0
 	github.com/spf13/cobra v1.8.0
 	go.uber.org/zap v1.26.0
-	k8s.io/client-go v0.29.4
+	k8s.io/client-go v0.30.0
```
Comment (rbtr) [inline go.mod:13]: blocking because we haven't greenlit k8s 1.30 yet

**Review by rbtr — CHANGES_REQUESTED** [review]

**Comment by rbtr** [comment]
@nddq can you dust this off and bump to k8s 1.30.1 like I did in https://github.com/microsoft/retina/pull/395?
critically we need controller runtime 0.18+

**Review by rbtr — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -1,14 +1,16 @@
 module github.com/microsoft/retina
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.2
```
Comment (rbtr) [inline go.mod:5]: actually can you revert this?
I don't really understand when the toolchain directive gets added, but I think this should remain `go 1.22` so we use the one in the builder image. If we have to have toolchain for some reason, it needs to be at least `1.22.3` for CVE fixes.

**Review by rbtr — CHANGES_REQUESTED** [review]

File: go.mod [hunk 3]
```diff
@@ -1,14 +1,16 @@
 module github.com/microsoft/retina
 
-go 1.22
+go 1.22.0
```
Comment (rbtr) [inline go.mod:3]: this is the one that needs to be `1.22`

Comment (nddq) [reply go.mod:3]: so we leave the toolchain as it is?

Comment (rbtr) [reply go.mod:3]: should be able to remove toolchain

**Review by rbtr — APPROVED** [review]

**Pull request merged**